app/src/main/java/space/taran/arknavigator/mvp/presenter/ResourcesPresenter.kt

```diff
@@ -113,16 +119,74 @@ class ResourcesPresenter(
             tagsSelectorPresenter.calculateTagsAndSelection()
 
             val path = (rootAndFav.fav ?: rootAndFav.root)
-            val title = if (path != null) "$path, " else ""
+            val title = if (path != null) "${path.last()}, " else ""
 
             viewState.setToolbarTitle("$title${roots.size} of roots chosen")
             viewState.setProgressVisibility(false)
         }
     }
 
-    override fun onDestroy() {
-        Log.d(RESOURCES_SCREEN, "destroying ResourcesPresenter")
-        super.onDestroy()
+    fun onMoveSelectedResourcesClicked(
+        directoryToMove: Path
+    ) = presenterScope.launch(Dispatchers.IO) {
+        withContext(Dispatchers.Main) {
+            viewState.setProgressVisibility(true, "Moving")
+        }
+        val resourcesToMove = gridPresenter.resources.filter { it.isSelected }
+        val results = resourcesToMove.map { item ->
+            async {
```

Comment:
Am I right, that would spawn a coroutine for each resource?

Reply:
Yes, we spawn a coroutine for each resource. The only performance limit is the number of threads in the pool, since the operations block the thread.
Coroutines are really lightweight. We can easily spawn 100,000 coroutines.
https://kotlinlang.org/docs/coroutines-basics.html#coroutines-are-light-weight